### PR TITLE
Update libraries

### PR DESCRIPTION
**Pull request by peppy**
Mainly to bring OTD up-to-date with upstream fixes.

### CODE REVIEW

File: osu.Framework/osu.Framework.csproj [hunk 1]
```diff
@@ -24,18 +24,22 @@
     <PackageReference Include="managed-midi" Version="1.9.14" />
     <PackageReference Include="Markdig" Version="0.22.1" />
     <PackageReference Include="FFmpeg.AutoGen" Version="4.3.0.1" />
-    <PackageReference Include="Microsoft.Extensions.ObjectPool" Version="5.0.3" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.9.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.MSBuild" Version="3.9.0">
+      <NoWarn>NU1701</NoWarn> <!-- Requires .NETFramework for MSBuild, but we use Microsoft.Build.Locator which allows this package to work in .NETCoreApp. -->
+    </PackageReference>
```
Comment (smoogipoo) [inline osu.Framework/osu.Framework.csproj:30]: Why did you move these?

Comment (peppy) [reply osu.Framework/osu.Framework.csproj:30]: Dunno, rider did. Is there an issue?

Comment (smoogipoo) [reply osu.Framework/osu.Framework.csproj:30]: At least `Microsoft.CodeAnalysis.Workspaces.MSBuild` is incompatible with netstandard so it feels wrong to keep it here, also with the comment being what it is. The only reason it works is because `Microsoft.Build.Locator` is used which is only compatible with netcore, and remains in the original location.

Comment (peppy) [reply osu.Framework/osu.Framework.csproj:30]: Right, it was inside some conditional. I guess rider ignores those?

**Review by smoogipoo — APPROVED** [review]
